### PR TITLE
mozc: new recipe

### PR DESCRIPTION
**Pull request by hanya**

### CODE REVIEW

File: app-i18n/mozc/mozc-2.20.2677.102.recipe [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+This is open source version."
+HOMEPAGE="https://github.com/google/mozc/"
+COPYRIGHT="2010-2016, Google Inc."
+LICENSE="BSD (3-clause)"
+REVISION="1"
+fileVersion="v2.20.2677.102.3"
+SOURCE_URI="https://github.com/hanya/mozc/releases/download/$fileVersion/mozc-$fileVersion.tar.bz2"
+CHECKSUM_SHA256="5b7c8c1e114b99221a2d9cf650fadd1cad9137986c94762f490a62af1048a8b1"
+SOURCE_DIR="mozc"
+if [ $effectiveTargetArchitecture == "x86" ]; then
```
Comment (waddlesplash) [inline app-i18n/mozc/mozc-2.20.2677.102.recipe]: Is the patch not needed on x86_64?

File: app-i18n/mozc/mozc-2.20.2677.102.recipe [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    PATCHES="mozc-$portVersion-x86.patch"
+fi
+SOURCE_URI_2="https://jaist.dl.sourceforge.net/project/zinnia/zinnia-tomoe/0.6.0-20080911/zinnia-tomoe-0.6.0-20080911.tar.bz2"
+CHECKSUM_SHA256_2="fd1a264b8aeea0ebec59e8c8765864a9543a75fb2dba349745bb3a3587790927"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+
+PROVIDES="
+    mozc
```
Comment (waddlesplash) [inline app-i18n/mozc/mozc-2.20.2677.102.recipe]: All indentation should be by 1 tab, not spaces.

File: app-i18n/mozc/mozc-2.20.2677.102.recipe [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    lib:libQt5Core
+    lib:libQt5Gui
+    lib:libQt5Widgets
+    "
+BUILD_REQUIRES="
+    haiku_devel
+    cmd:python
+    cmd:ninja
+    protobuf
+    cmd:protoc
```
Comment (waddlesplash) [inline app-i18n/mozc/mozc-2.20.2677.102.recipe]: All `cmd:` BUILD requirements go in the _PREREQUIRES section.

File: app-i18n/mozc/mozc-2.20.2677.102.recipe [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    protobuf_devel
+    lib:libprotobuf
```
Comment (waddlesplash) [inline app-i18n/mozc/mozc-2.20.2677.102.recipe]: You shouldn't need this; protobuf_devel should pull it in automatically. (And replacing `protobuf_devel` with `devel:libprotobuf` would be preferred, too.)

File: app-i18n/mozc/mozc-2.20.2677.102.recipe [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    devel:libGL
+    cmd:pkg_config
+    cmd:which
+    cmd:xres
+    "
+BUILD_PREREQUIRES="
+    cmd:python
+    cmd:g++
+    cmd:gcc
+    gyp
```
Comment (waddlesplash) [inline app-i18n/mozc/mozc-2.20.2677.102.recipe]: Should be `cmd:gyp`.

File: app-i18n/mozc/mozc-2.20.2677.102.recipe [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    cp $MozcOutDir/mozc_server $MozcDataDir/mozc_server
+    cp $MozcOutDir/mozc_tool $MozcDataDir/mozc_tool
+    cp $MozcOutDir/handwriting-ja.model \
+            $MozcDataDir/zinnia/handwriting-ja.model
+    cp $MozcOutDir/mozc_emacs_helper $MozcDataDir/mozc_emacs_helper
+    cp src/data/installer/credits_en.html $MozcDataDir/credits_en.html
+
+    cp $MozcOutDir/x-vnd.Mozc-InputMethod/* \
+       $dataDir/locale/catalogs/x-vnd.Mozc-InputMethod
+}
```
Comment (waddlesplash) [inline app-i18n/mozc/mozc-2.20.2677.102.recipe]: Missing EOL at end of file.

File: app-i18n/mozc/mozc-2.20.2677.102.recipe (hunk 1 above)
Comment (hanya) [inline app-i18n/mozc/mozc-2.20.2677.102.recipe]: Not required for x86_64. Mozc defines own int32 and uint32 which conflicts with defiend by Haiku on x86. But not on x86_64.

**Comment by hanya** [comment]
Thanks for the response. I've fixed the recipe according to advaices.

File: app-i18n/mozc/mozc-2.20.2677.102.recipe [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+BUILD_REQUIRES="
+	haiku_devel
+	devel:libprotobuf
+	devel:libQt5Core
+	devel:libQt5Gui
+	devel:libQt5Widgets
+	devel:libGL
+	"
+BUILD_PREREQUIRES="
+	cmd:python
```
Comment (korli) [inline app-i18n/mozc/mozc-2.20.2677.102.recipe]: Please choose a python version here, otherwise Haikuporter will pick a random one: python2 or python 3.

Comment (hanya) [inline app-i18n/mozc/mozc-2.20.2677.102.recipe]: I see. GYP can work only with Python2. Fixed.

**Comment by waddlesplash** [comment]
Thanks very much for the contribution!

**Pull request merged**